### PR TITLE
fix(broker/conf) change passwords in html forms

### PR DESCRIPTION
**Pull request by a-launois**
## Description

Hide passwords in broker configuration form

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 20.10.x
- [x] 21.04.x
- [x] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by callapa — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

File: www/class/centreonConfigCentreonBroker.php [hunk 1]
```diff
@@ -156,6 +156,9 @@ class CentreonConfigCentreonBroker
         <td><div class="ams">{label_3}</div>{selected}</td>
         </tr></table>{javascript}';
 
+    /** @var string */
+    private $passwordReplacementValue = '**********';
+
```
Comment (jeremyjaouen) [inline www/class/centreonConfigCentreonBroker.php:161]:
```suggestion
```

File: www/class/centreonConfigCentreonBroker.php [hunk 2]
```diff
@@ -835,6 +838,101 @@ public function updateConfig(int $id, array $values)
...
+     * @param int $configId
+     * @param array<string,mixed> $values
+     */
+    private function revealLuaPasswords(int $configId, array &$values): void
+    {
+        foreach ($values['output'] as &$output) {
+            foreach (array_keys($output) as $key) {
+                if (
+                    preg_match('/^lua_parameter__value_(\\d+)$/', (string) $key, $matches)
+                    && $output["lua_parameter__value_{$matches[1]}"] === $this->passwordReplacementValue
```
Comment (jeremyjaouen) [inline www/class/centreonConfigCentreonBroker.php:879]:
```suggestion
                    && $output["lua_parameter__value_{$matches[1]}"] === \CentreonAuth::PWS_OCCULTATION
```

File: www/class/centreonConfigCentreonBroker.php [hunk 3]
```diff
@@ -1075,19 +1180,30 @@ public function getForms($config_id, $tag, $page, $tpl)
                     $suffix = '';
                 }
                 $arrayMultipleValues[$fieldname]['suffix'] = $suffix;
-                $arrayMultipleValues[$fieldname]['values'][] = $row['config_value'];
+                $arrayMultipleValues[$fieldname]['values'][] =
+                    $isTypePassword && $suffix === 'value' ? $this->passwordReplacementValue : $row['config_value'];
```
Comment (jeremyjaouen) [inline www/class/centreonConfigCentreonBroker.php:1184]:
```suggestion
                    $isTypePassword && $suffix === 'value' ? \CentreonAuth::PWS_OCCULTATION : $row['config_value'];
```

File: www/class/centreonConfigCentreonBroker.php [hunk 4]
```diff
@@ -1075,19 +1180,30 @@ public function getForms($config_id, $tag, $page, $tpl)
...
             } else {
                 if (isset($formsInfos[$row['config_group_id']]['defaults'][$fieldname])) {
                     if (!is_array($formsInfos[$row['config_group_id']]['defaults'][$fieldname])) {
                         $formsInfos[$row['config_group_id']]['defaults'][$fieldname] = array(
                             $formsInfos[$row['config_group_id']]['defaults'][$fieldname]
                         );
                     }
-                    $formsInfos[$row['config_group_id']]['defaults'][$fieldname][] = $row['config_value'];
+                    $formsInfos[$row['config_group_id']]['defaults'][$fieldname][] =
+                        $row['config_key'] === 'db_password' ? $this->passwordReplacementValue : $row['config_value'];
```
Comment (jeremyjaouen) [inline www/class/centreonConfigCentreonBroker.php:1199]:
```suggestion
                        $row['config_key'] === 'db_password' ? \CentreonAuth::PWS_OCCULTATION : $row['config_value'];
```

File: www/class/centreonConfigCentreonBroker.php [hunk 5]
```diff
@@ -1075,19 +1180,30 @@ public function getForms($config_id, $tag, $page, $tpl)
...
                 } else {
-                    $formsInfos[$row['config_group_id']]['defaults'][$fieldname] = $row['config_value'];
+                    $formsInfos[$row['config_group_id']]['defaults'][$fieldname] =
+                        $row['config_key'] === 'db_password' ? $this->passwordReplacementValue : $row['config_value'];
```
Comment (jeremyjaouen) [inline www/class/centreonConfigCentreonBroker.php:1202]:
```suggestion
                        $row['config_key'] === 'db_password' ? \CentreonAuth::PWS_OCCULTATION : $row['config_value'];
```

File: www/class/centreonConfigCentreonBroker.php [hunk 6]
```diff
@@ -1075,19 +1180,30 @@ public function getForms($config_id, $tag, $page, $tpl)
...
                     $formsInfos[$row['config_group_id']]['defaults'][$fieldname . '[' . $row['config_key'] . ']'] =
-                        $row['config_value']; // Radio button
+                        $row['config_key'] === 'db_password'
+                        ? $this->passwordReplacementValue
```
Comment (jeremyjaouen) [inline www/class/centreonConfigCentreonBroker.php:1205]:
```suggestion
                        ? \CentreonAuth::PWS_OCCULTATION
```

File: www/class/centreonConfigCentreonBroker.php (hunk 2 above)
Comment (jeremyjaouen) [reply www/class/centreonConfigCentreonBroker.php:879]: To be consistent with all the password_occultation we've created a constant 
https://github.com/centreon/centreon/blob/develop/www/class/centreonAuth.class.php#L45

Comment (callapa) [reply www/class/centreonConfigCentreonBroker.php:879]: @jeremyjaouen Is it possible to use the contants of classes that do not come from the legacy?

Comment (a-launois) [reply www/class/centreonConfigCentreonBroker.php:879]: It works, but should I use it or should we keep the legacy separate ?

**Review by kduret — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**